src/main/java/it/pagopa/pm/gateway/controller/CreditCardPaymentController.java

```diff
@@ -80,15 +82,16 @@ private ResponseEntity<StepZeroResponse> buildResponseStep0(StepZeroResponse ste
     }
 
     @PostMapping(REQUEST_PAYMENTS_RESUME_METHOD)
-    public ResponseEntity<String> resumeCreditCardPayment(@RequestHeader(required = false, value = MDC_FIELDS) String mdcFields,
-                                                          @PathVariable String requestId,
-                                                          @RequestBody CreditCardResumeRequest request) {
+    public ResponseEntity<VposResumeMethodResponse> resumeCreditCardPayment(@RequestHeader(required = false, value = MDC_FIELDS) String mdcFields,
+                                                                            @PathVariable UUID requestId,
+                                                                            @RequestBody CreditCardResumeRequest request) {
         log.info("START - POST {}{} info for requestId: {}", REQUEST_PAYMENTS_VPOS, REQUEST_PAYMENTS_RESUME_METHOD, requestId);
         MdcUtils.setMdcFields(mdcFields);
-        resumeStep1Service.startResumeStep1(request, requestId);
+        VposResumeMethodResponse response = new VposResumeMethodResponse(String.valueOf(requestId));
+        resumeStep1Service.startResumeStep1(request, String.valueOf(requestId));
```

Comment:
(in general) `valueOf` maps null values to `"null"`. Given that this is a path variable (so you can safely expect it to never be null), prefer `toString`.

Reply:
https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/200/commits/97e806ae8f4d63c8281711d220999374caf7024b